### DISCUSSION TITLE
MergeGraph breaks LazyGetPropertyAsync logic

### DISCUSSION THREAD

**Discussion by lovror** [MarimerLLC/csla#3293] [Questions] [answered]
When saving Csla.Blazor.ViewModel using SaveAsync before loading all lazy loaded properties LazyGetPropertyAsync logic is broken and the viewmodel can't be used for further editing. 
Result returned from ViewModels save method is feeded to MergeGraph method which calls LoadData for null values and sets FieldManager.FieldExists to true for lazy loaded properties that weren't really not loaded yet.

One workaround is to override DoSaveAsync method on ViewModel to provide custom Model save behavior and use a customized MergeGraph method.
Another option is to alter MergeGraph method and prevent loading null values in case of lazy loaded properties.

```
if ((item.RelationshipType & RelationshipTypes.PrivateField) == RelationshipTypes.PrivateField)
    Csla.Reflection.MethodCaller.CallPropertySetter(target, item.Name, sourceValue);
else if (sourceValue != null || (item.RelationshipType & RelationshipTypes.LazyLoad) != RelationshipTypes.LazyLoad) 
    LoadProperty(target, item, sourceValue);

```
Am I missing something and using ViewModel wrong or is this issue appropriate for pull request?

Something similar was discussed here https://github.com/MarimerLLC/csla/discussions/3213 but no solution was provided.

**Comment by rockfordlhotka** [comment, marked as answer]
Looking back, this sort of bug was _attempted_ to be fixed here: https://github.com/MarimerLLC/csla/pull/2900

I'm guessing this is a variation on the issue that wasn't caught at that time.

**Reply by rockfordlhotka** [reply]
I've logged this new bug as #3294.

If you'd like to submit a PR to fix it that is excellent!

**Reply by lovror** [reply]
Thank you for your prompt answer. I will create a PR on main branch to fix this.
